tests/test_filters.py

```diff
@@ -156,6 +156,38 @@ class SearchListView(generics.ListAPIView):
 
         reload_module(filters)
 
+    def test_search_with_filter_subclass(self):
+        class CustomSearchFilter(filters.SearchFilter):
+            # Filter that dynamically changes search fields
+            def get_search_fields(self, view, request):
+                if request.query_params.get('title_only'):
+                    return ('$title',)
+                return super(CustomSearchFilter, self).get_search_fields(
+                    view, request)
+
+            # Ensures that the base class implements get_search_terms
```

Comment:
As an aside, this is not strictly true. If `SearchFilter` is changed to no longer call `get_search_terms()`, nothing guarantees that this method is invoked. I would recommend removing `get_search_terms()` here.